lib/kube/proxy/forwarder.go

```diff
@@ -2068,15 +2067,42 @@ func (f *Forwarder) findKubeClusterByName(name string) (types.KubeCluster, error
 	return nil, trace.NotFound("cluster %s not found", name)
 }
 
-// newKubernetesClusterV3FromDetails copies the details.kubeCluster and populates it with the
-// kube_service's static and dynamic labels values.
+// newKubernetesClusterV3FromDetails clones the details.kubeCluster and replaces the cluster
+// dynamic labels with their latest value.
 func (f *Forwarder) newKubernetesClusterV3FromDetails(details *kubeDetails) (*types.KubernetesClusterV3, error) {
 	clonedCluster := details.kubeCluster.Copy()
-	clonedCluster.Metadata.Labels = f.getClusterStaticLabels(clonedCluster)
```

Comment:
Why are we not setting static labels here anymore but still setting dynamic labels? Seems confusing.